### DISCUSSION TITLE
Artifacts in rtsp streams, not in wyze app stream

### DISCUSSION THREAD

**Discussion by baudneo** [gtxaspec/wz_mini_hacks#221] [Q&A]
First, thank you for your work! Amazing!

I bought a v3 to try out this FOSS project and see how I could use it to integrate wyze with zoneminder. I am the AI/ml python dev for zoneminder ATM. I had a v2 with wyze rtsp firmware that I had integrated with zoneminder and it worked well. 

I put the v3 in the same spot as the v2 last night and I have been getting alot of artifacts/tearing which sets off a motion event in zoneminder which then sets off the object detection pipeline. The tearing/artifacts are only on the rtsp stream as the wyze app is showing a clear stream.

I read up on open and closed issues and this discussion section but could not really find any advice or remedies for the issue. Does anyone have any ideas I could try out?

Thanks in advance!

Attached is a screenshot.

![Screenshot_20220805-230603_zmNinja.png](https://user-images.githubusercontent.com/86508179/183254831-0f0fc0ef-8ec8-4d70-8c46-26ccce23b99f.png)

**Comment by clee604** [comment]
Hi I had similar issues and honestly I'm not sure if it got resolved completely but recently I don't notice it as much. I would try the local_dns and fsck options as those seemed to have helped.

Particularly the dns option stopped the dns flood issue I was having and I think it was impacting the rtsp stream evem tho the camera was hardwired

**Reply by baudneo** [reply]
I will give that a try. Thanks!

**Reply by baudneo** [reply]
So I narrowed the issue down. ZM decodes RTSP stream usinf FFMPEG. FFMPEG was configured to use CUDA HW decoder. CUDA was causing the tearing as I am now only decoding using CPU and the stream is working fine. Now we get to chase down the FFMPEG issues....

**Reply by guarddog13** [reply]
> So I narrowed the issue down. ZM decodes RTSP stream usinf FFMPEG. FFMPEG was configured to use CUDA HW decoder. CUDA was causing the tearing as I am now only decoding using CPU and the stream is working fine. Now we get to chase down the FFMPEG issues....

I have this exact issue using AgentDVR. With AgentDVR it happens on both CPU and GPU decoder. I've lowered the framerate of the streams. I've set it to udp, tcp, udp multi-stream. I have dnsmasq and fsck turned on. What seemed to help it the best was when I switched the dnsresolv ips to google..(my network is not a fan of cloudfare). I might try another dns upstream and see that helps...maybe set it to my adguard and give it full access?
